### PR TITLE
interp: add ClampedCubic interpolator

### PR DESCRIPTION
**Pull request by romanwerpachowski**
Please take a look.

### CODE REVIEW

File: interp/cubic_test.go [hunk 1]
```diff
@@ -800,5 +800,37 @@ func TestNaturalCubicFit(t *testing.T) {
...
+}
 
+func TestClampedCubicFit(t *testing.T) {
+	t.Parallel()
+	xs := []float64{-1, 0, 2, 3.5}
+	ys := []float64{2, 0, 2, 1.5}
+	var cc ClampedCubic
+	err := cc.Fit(xs, ys)
+	if err != nil {
+		t.Errorf("Error when fitting NaturalCubic: %v", err)
```
Comment (vladimir-ch) [inline interp/cubic_test.go:812]: ClampedCubic?

Comment (romanwerpachowski) [reply interp/cubic_test.go]: Thanks, fixed.

**Comment by romanwerpachowski** [comment]
PTAL.

**Comment by soypat** [comment]
Where do the `want` values come from? Could we not make property based tests?

**Comment by romanwerpachowski** [comment]
They come from `scipy.interpolate.CubicSpline`, which I treat as a trusted reference implementation until proven otherwise :) I could test properties of `ClampedCubic` directly, but it's more work and the current test is sufficient IMHO.

**Comment by romanwerpachowski** [comment]
*ping*

**Review by vladimir-ch — APPROVED** [review]

**Pull request merged**